pinot-common/src/main/java/org/apache/pinot/common/utils/ServiceStatus.java

```diff
@@ -227,70 +230,98 @@ private boolean isDone() {
     @Override
     public synchronized Status getServiceStatus() {
 
+      String descriptionSuffix = "";
       while (!isDone()) {
         String resourceName;
         if (_resourceIterator == null || !_resourceIterator.hasNext()) {
           _resourceIterator = _resourcesToMonitor.iterator();
         }
         resourceName = _resourceIterator.next();
-        IdealState idealState = getResourceIdealState(resourceName);
+        descriptionSuffix = String
+            .format("waitingFor=%s, resource=%s, numResourcesLeft=%d, numTotalResources=%d, minStartCount=%d,", getMatchName(),
+                resourceName, _resourcesToMonitor.size(), _numTotalResourcesToMonitor, _minResourcesStartCount);
 
-        // If the resource has been removed or disabled, ignore it
-        if (idealState == null || !idealState.isEnabled()) {
+        Pair<Status, String> statusPair = evaluateResourceStatus(resourceName, descriptionSuffix);
```

Comment:
<nit> might be better to append the suffix after the method is called instead of passing the suffix to it; just seems cleaner </nit>